### PR TITLE
Location changes delegate methods.

### PR DESCRIPTION
**Pull request by fabian-guerra**
Adds delegate methods to let developers know about location accuracy changes.

### CODE REVIEW

File: platform/darwin/src/MGLLocationManager.m [hunk 1]
```diff
@@ -38,7 +38,15 @@ - (CLLocationAccuracy)desiredAccuracy {
...
 - (CLAuthorizationStatus)authorizationStatus {
+#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 140000
+    if (@available(iOS 14.0, *)) {
+        return self.locationManager.authorizationStatus;
+    } else {
+        return kCLAuthorizationStatusNotDetermined;
+    }
+#else
     return [CLLocationManager authorizationStatus];
+#endif
```
Comment (neelmistry94) [inline platform/darwin/src/MGLLocationManager.m:49]: I'm not sure if this stylistic or a necessity, but do we need the pre processor #if statements?

File: platform/darwin/src/MGLLocationManager.m [hunk 2]
```diff
@@ -38,7 +38,15 @@ - (CLLocationAccuracy)desiredAccuracy {
...
 }
 
 - (void)setActivityType:(CLActivityType)activityType {
```
Comment (neelmistry94) [inline platform/darwin/src/MGLLocationManager.m:52]: I know you didn't change this, but the function says set activity type, but you are just returning an existing activity type, that doesn't make sense

File: platform/darwin/src/MGLLocationManager.m [hunk 3]
```diff
@@ -49,6 +57,22 @@ - (CLActivityType)activityType {
...
+    if (@available(iOS 14.0, *)) {
+        return self.locationManager.accuracyAuthorization;
+    } else {
+        return CLAccuracyAuthorizationFullAccuracy;
+    }
+}
+
+- (void)requestTemporaryFullAccuracyAuthorizationWithPurposeKey:(NSString *)purposeKey {
+    if (@available(iOS 14.0, *)) {
+        [self.locationManager requestTemporaryFullAccuracyAuthorizationWithPurposeKey:purposeKey];
```
Comment (neelmistry94) [inline platform/darwin/src/MGLLocationManager.m:71]: wouldn't this be an endless loop/recursive call if you are on iOS14. 
This looks like the implementation is calling the method itself over and over again, and if you are on iOS 14, that will happen non stop

File: platform/darwin/src/MGLLocationManager.m [hunk 4]
```diff
@@ -112,5 +136,9 @@ - (void)locationManager:(CLLocationManager *)manager didFailWithError:(NSError *
         [self.delegate locationManager:self didFailWithError:error];
     }
 }
-
+#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 140000
+- (void)locationManagerDidChangeAuthorization:(CLLocationManager *)manager {
+    [self.delegate locationManagerDidChangeAuthorization:self];
```
Comment (neelmistry94) [inline platform/darwin/src/MGLLocationManager.m:141]: same comment as above

File: platform/darwin/src/MGLLocationManager.m (hunk 1 above)
Comment (fabian-guerra) [reply platform/darwin/src/MGLLocationManager.m:49]: Pragma marks are there to make it compatible with Xcode 11.

File: platform/darwin/src/MGLLocationManager.m (hunk 2 above)
Comment (fabian-guerra) [reply platform/darwin/src/MGLLocationManager.m:52]: The default implementation of `MGLLocationManager` is a wrapper of `CLLocationManager.

File: platform/darwin/src/MGLLocationManager.m (hunk 3 above)
Comment (fabian-guerra) [reply platform/darwin/src/MGLLocationManager.m:71]: Same comment as above.

File: platform/darwin/src/MGLLocationManager.m (hunk 4 above)
Comment (fabian-guerra) [reply platform/darwin/src/MGLLocationManager.m:141]: Same as https://github.com/mapbox/mapbox-gl-native-ios/pull/376#discussion_r472232475

File: platform/ios/src/MGLMapViewDelegate.h [hunk 5]
```diff
@@ -17,7 +17,19 @@ NS_ASSUME_NONNULL_BEGIN
...
+ 
+ This method is called after the user changes location accuracy authorization when
+ requesting location permissions or in privacy settings.
+ 
+ @param mapView The map view that changed its location accuracy authorization.
+ @param accuracyAuthorization The new accuracy authorization value.
+ 
+ */
+- (void)mapView:(MGLMapView *)mapView didChangeAccuracyAuthorization:(CLAccuracyAuthorization)accuracyAuthorization;
+#endif
```
Comment (nishant-karajgikar) [inline platform/ios/src/MGLMapViewDelegate.h:32]: We shouldn't need the xcode 11 compatibility test here as this is just a wrapper.

Also -- shouldn't this method be optional? There can be consumers of the map who don't care about user location entirely.

File: platform/ios/app/MBXOrnamentsViewController.m [hunk 6]
```diff
@@ -64,7 +64,7 @@ - (void)viewDidLoad {
                        zoomLevel:16
                        direction:30
                         animated:NO];
-    mapView.delegate = self;
+    
     mapView.showsScale = YES;
```
Comment (nishant-karajgikar) [inline platform/ios/app/MBXOrnamentsViewController.m:68]: Why are we removing this?

File: platform/ios/src/MGLMapViewDelegate.h (hunk 5 above)
Comment (fabian-guerra) [reply platform/ios/src/MGLMapViewDelegate.h:32]: > We shouldn't need the xcode 11 compatibility test here as this is just a wrapper.

In order to build the SDK with either Xcode 11/12 this has to be here. Unless we drop Xcode 11 support.

> Also -- shouldn't this method be optional? There can be consumers of the map who don't care about user location entirely.

Right. However with the privacy changes this has to be required to let developers be aware that they will have to handle that, even tho they don't use location updates.

File: platform/ios/app/MBXOrnamentsViewController.m (hunk 6 above)
Comment (fabian-guerra) [reply platform/ios/app/MBXOrnamentsViewController.m:68]: This was dead code in the examples that surfaced with the introduction of the new delegate method.

**Comment by nishant-karajgikar** [comment]
Thanks for making the change, @fabian-guerra . I won't be able to test this on an iOS14 device just yet (running into trouble downloading the latest xcode beta). Feel free to merge this and we can address anything we see during test in a subsequent PR.

**Pull request merged**